service/internal/cloudtags/spec.go

```diff
@@ -0,0 +1,10 @@
+package cloudtags
+
+import "context"
+
+type Interface interface {
+	// AreClusterTagsEquals compares current cluster tags with the stack tags
+	AreClusterTagsEquals(ctx context.Context, ctags map[string]string, stags map[string]string) bool
```

Comment:
Suggestions, the name is more idiomatic that way and the interface is simpler, hiding the complexity of having to deal with CRs and Cloud Formation tags. 

```suggestion
	ClusterTagsEqual(ctx context.Context, clusterID string) (bool, error)
```

Reply:
At the beginning I wrote it like `AreClusterTagsEquals(ctx context.Context, clusterID, stags map[string]string) bool`
but if we dont inject stack tags then you need to query again AWS API to get those (which we want to avoid). It  would great in terms of hiding complexity but we already call AWS API a lot and I did not want add other extra call (having the tags already on place)

Reply:
That is what the caching is for, so that it does not matter how often we want to fetch or use the data within a single reconciliation loop.

Reply:
ok let me take a look to caching code and I will do a PR targeting this branch